src/Stateless/StateConfiguration.Async.cs

```diff
@@ -201,7 +201,6 @@ public StateConfiguration InternalTransitionAsync<TArg0>(TriggerWithParameters<T
             /// <returns>The receiver.</returns>
             public StateConfiguration OnActivateAsync(Func<Task> activateAction, string activateActionDescription = null)
             {
-                Enforce.ArgumentNotNull(activateAction, nameof(activateAction));
```

Comment:
Ah, I see - the check is performed in _representation.AddActivateAction(). Works for me! 👍